src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1512,9 +1512,13 @@ private void WriteRedRewriter()
                             {
                                 Write("VisitToken(node.{1})", CamelCase(field.Name), field.Name);
                             }
+                            else if (IsOptional(field))
+                            {
+                                Write("({1})Visit(node.{2})", CamelCase(field.Name), GetFieldType(field, green: false), field.Name);
+                            }
                             else
                             {
-                                Write("({1})Visit(node.{2})", CamelCase(field.Name), field.Type, field.Name);
+                                Write("({1})Visit(node.{2}) ?? throw new ArgumentNullException(\"{0}\")", CamelCase(field.Name), GetFieldType(field, green: false), field.Name);
```

Comment:
This is technically a breaking change. I think I'm fine with it, but we need to be on the alert for new `ArgumentNullException`s coming from syntax rewriters and be prepared to revert if it's widespread.

Reply:
This came in through Sam's change.  I am preserving that behavior.